pkg/solana/chainreader/chain_reader.go

```diff
@@ -223,12 +223,12 @@ func (s *SolanaChainReaderService) CreateContractType(readIdentifier string, for
 func (s *SolanaChainReaderService) init(namespaces map[string]config.ChainReaderMethods) error {
 	for namespace, methods := range namespaces {
 		for methodName, method := range methods.Methods {
-			var idl codec.IDL
+			var idl solanacodec.IDL
 			if err := json.Unmarshal([]byte(method.AnchorIDL), &idl); err != nil {
 				return err
 			}
 
-			idlCodec, err := codec.NewIDLAccountCodec(idl, config.BuilderForEncoding(method.Encoding))
+			idlCodec, err := solanacodec.NewIDLAccountCodec(idl, config.BuilderForEncoding(method.Encoding))
```

Comment:
Will replace these in another PR to initialise Codec in the same way like on EVM